libbeat/otelbeat/oteltest/oteltest.go

```diff
@@ -0,0 +1,115 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Package oteltest provides test utilities for OpenTelemetry and Beats components.
+package oteltest
+
+import (
+	"bytes"
+	"context"
+	"sync"
+	"testing"
+
+	"github.com/elastic/elastic-agent-libs/mapstr"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type CheckMultipleReceiversParams struct {
+	T *testing.T
+	// Factory that allows to create a receiver.
+	Factory receiver.Factory
+	// Config of the receiver to use.
+	Config component.Config
+	// AssertFunc is a function that asserts the test conditions.
+	AssertFunc func(t *testing.T, logs map[string][]mapstr.M)
+}
+
+// CheckMultipleReceivers checks that multiple receivers can be created and started
+// on the same process without errors.
+func CheckMultipleReceivers(params CheckMultipleReceiversParams) {
+	t := params.T
+	var logsMu sync.Mutex
+	logs := make(map[string][]mapstr.M)
+
+	ctx := context.Background()
+	createReceiver := func(t *testing.T, name string) receiver.Logs {
+		t.Helper()
+
+		var zapLogs bytes.Buffer
+		core := zapcore.NewCore(
+			zapcore.NewJSONEncoder(zap.NewProductionEncoderConfig()),
+			zapcore.AddSync(&zapLogs),
+			zapcore.DebugLevel)
+
+		receiverSettings := receiver.Settings{}
+		receiverSettings.Logger = zap.New(core).Named(name)
+
+		logConsumer, err := consumer.NewLogs(func(ctx context.Context, ld plog.Logs) error {
+			for i := 0; i < ld.ResourceLogs().Len(); i++ {
+				rl := ld.ResourceLogs().At(i)
+				for j := 0; j < rl.ScopeLogs().Len(); j++ {
+					sl := rl.ScopeLogs().At(j)
+					for k := 0; k < sl.LogRecords().Len(); k++ {
+						log := sl.LogRecords().At(k)
+						logsMu.Lock()
+						logs[name] = append(logs[name], log.Body().Map().AsRaw())
+						logsMu.Unlock()
+						t.Logf("ingested log for %q: %v", name, log.Body().Map().AsRaw())
```

Comment:
Do we really need to log during the test?

I believe it is better for tests not to log anything and instead collect any information that might be required to debug and log them all at once if the test fails. This makes the, successful, test output less verbose.

Reply:
Agree, we already print the zap logs in case of failure, I just removed printing the logs.